spec/active_force/sobject/includes_spec.rb

```diff
@@ -285,6 +285,246 @@ module ActiveForce
           expect(account.owner.id).to eq '321'
         end
       end
+
+      context 'when invalid associations are passed' do
+        it 'raises an error' do
+          expect { Quota.includes(:invalid).find('123') }.to raise_error(ActiveForce::Association::InvalidAssociationError, 'Association named invalid was not found on Quota')
+        end
+      end
+    end
+
+    describe '.includes with nested associations' do
+
+      context 'with custom objects' do
+        it 'formulates the correct SOQL query' do
+          soql = Quota.includes(prez_clubs: :club_members).where(id: '123').to_s
+          expect(soql).to eq "SELECT Id, Bar_Id__c, (SELECT Id, QuotaId, (SELECT Id, Name, Email FROM ClubMembers__r) FROM PrezClubs__r) FROM Quota__c WHERE (Bar_Id__c = '123')"
```

Comment:
Optional: Could possibly use heredocs for these query strings + `.squish` to make it more readable.  Something like:

```ruby
<<~SOQL.squish
  SELECT Id, Bar_Id__c,
    (SELECT Id, QuotaId,
      (SELECT Id, Name, Email FROM ClubMembers__r)
    FROM PrezClubs__r)
  FROM Quota__c
  WHERE (Bar_Id__c = '123')
SOQL
```

Reply:
never knew about `.squish` but looks nice, would update